R/archivePackages.R

```diff
@@ -10,58 +10,84 @@
 ##' defaults to the value of the \dQuote{dratRepo} option with
 ##' \dQuote{"~/git/drat"} as fallback
 ##' @param type Character variable for the type of repository, so far
-##'  \dQuote{source}, \dQuote{win.binary}, \dQuote{mac.binary}, 
+##'  \dQuote{source}, \dQuote{win.binary}, \dQuote{mac.binary},
 ##'  \dQuote{mac.binary.mavericks} or \dQuote{mac.binary.el-capitan}
 ##' @param pkg Optional character variable specifying a package name(s), whose
-##' older versions should be archived. If missing (the default), archiving is 
+##' older versions should be archived. If missing (the default), archiving is
 ##' performed on all packages.
-##' @param version R version information in the format \code{X.Y} or 
+##' @param version R version information in the format \code{X.Y} or
 ##'   \code{X.Y.Z}. Only used, if archiving binary packages.
-##'   (default: \code{version = getRversion()})
+##'   By default all R versions found in the respective CRAN-like directory
+##'   for the chosen `type` are checked.
 ##' @examples
 ##' \dontrun{
 ##'   archivePackages()   # archive all older package versions
 ##'   archivePackages(pkg = "drat")  # archive older copies of just one package
 ##' }
 ##' @author Thomas J. Leeper
 archivePackages <- function(repopath = getOption("dratRepo", "~/git/drat"),
-                            type = "source", 
-                            pkg,
-                            version = getRversion()) {
+                            type = c(
+                                "source", "mac.binary", "mac.binary.el-capitan",
+                                "mac.binary.mavericks", "win.binary"
+                            ),
+                            pkg = NULL,
+                            version = NULL) {
     
-    ## knows how to handle binary repos
-    repodir <- contrib.url2(repopath, type, version)
-
-    archive <- file.path(repodir, "Archive")
-    if (!file.exists(archive)) {
-        if (!dir.create(archive, recursive = TRUE)) {
-            stop("Archive directory not found and couldn't be created\n", call. = FALSE)
+    for (type in type) {
+        
+        # if no R version is supplied, the one from the
+        # respective "type" directory is scraped
+        if (is.null(version)) {
+            if (grepl("binary", type)) {
+                dirs = list.dirs(gsub(
+                    "/[0-9].[0-9]$", "", contrib.url(repopath, type)), 
+                    recursive = FALSE)
+                r_versions = substring(dirs, regexpr("[0-9].[0-9]$", dirs))
+            } else {
+                # no version for source packages
+                r_versions <- ""
+            }
+        } else {
+            dirs = list.dirs(gsub(
+                "/[0-9].[0-9]$", "", contrib.url(repopath, type)), 
+                recursive = FALSE)
+            r_versions = substring(dirs, regexpr("[0-9].[0-9]$", dirs))
         }
```

Comment:
Still not my style of writing but then I just like others think how R Core writes code matters most, and i happen to use the same editor they use (or at least a number of them).   So whatevs.   Thanks for adhering to four space indentation though, that is appreciated.  (And this repo is so small that I never bothered with `,editoconfig` which may be overkill.)